arelle/plugin/validate/HMRC/__init__.py

```diff
@@ -168,18 +168,23 @@ def validateXbrlStart(val, parameters=None, *args, **kwargs):
     val.txmyType = None
     for doc in val.modelXbrl.modelDocument.referencesDocument:
         ns = doc.targetNamespace
-        if ns:
-            if ns.startswith("http://www.xbrl.org/uk/char/"): val.txmyType = "charities"
-            elif ns.startswith("http://www.xbrl.org/uk/gaap/"): val.txmyType = "ukGAAP"
-            elif ns.startswith("http://www.xbrl.org/uk/ifrs/"): val.txmyType = "ukIFRS"
-            elif ns.startswith(FRC_URL_DOMAIN):
-                if any((concept.qname.namespaceURI.startswith(FRC_URL_DOMAIN) and concept.modelDocument.inDTS)
-                       for concept in val.modelXbrl.nameConcepts.get("AccountsType", ())):
+        if not ns:
+            continue
+        if ns.startswith("http://www.xbrl.org/uk/char/"):
+            val.txmyType = "charities"
+        elif ns.startswith("http://www.xbrl.org/uk/gaap/"):
+            val.txmyType = "ukGAAP"
+        elif ns.startswith("http://www.xbrl.org/uk/ifrs/"):
+            val.txmyType = "ukIFRS"
+        elif ns.startswith(FRC_URL_DOMAIN):
+            val.txmyType = "FRS"
+            for concept in val.modelXbrl.nameConcepts.get("AccountsType", ()):
+                if concept.qname.namespaceURI.startswith(FRC_URL_DOMAIN) and concept.modelDocument.inDTS:
                     val.txmyType = "FRS-2022"
-                else:
-                    val.txmyType = "FRS"
-            else: continue
-            break
+                    break
```

Comment:
this break seems wrong

Reply:
It breaks the inner for loop on 181 if a `val.txmyType` is set?

Reply:
👍 right